tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1945,6 +1945,13 @@ def stdin_publish(pub, msg_class, rate, once, filename, verbose):
     else:
         iterator = stdin_yaml_arg
 
+    exactly_one_message = False
+    for _ in iterator():
+        if exactly_one_message:
+            exactly_one_message = False
+            break
+        exactly_one_message = True
```

Comment:
Doesn't this alter the content returned by the iterator in the later loop - in the case of the iterator being `stdin_yaml_arg`?

Reply:
Yes, that's right. The fix makes not much sense for stdin stream anyway, so I limited it to actual files now.